docs/website/root/mithril/threat-model.md

```diff
@@ -0,0 +1,318 @@
+
+## Assets
+
+For each asset we first identify what protection is required: Availability, Confidentiality, Integrity ie. the [CIA Triad](https://www.splunk.com/en_us/blog/learn/cia-triad-confidentiality-integrity-availability.html)
+
+:::info Note
+
+I follow Pascal's suggestion to identify assets as processes rather than particular resources, except in the case of actual pieces of data (eg. keys). The latter should be listed in the threats section and their protection mitigated
+
+:::
+
+We then identify threats and countermeasures
+
+#### KES private keys
+
+* KES key is present only on BP Node but needs to be shared with both the cardano-node process and the mithril-signer process
+* KES keys are needed by mithril-signer in order to sign verification key along with an operational certificate which authenticates the key from this stake pool id
+* This signing happens at every epoch
+    * **availability**: Yes (If KES key is unavailable then signing cannot proceeed)
+    * **confidentiality**: Yes (Capturing KES private keys allow an attacker to impersonate a registered SPO on-chain and produce blocks on his behalf)
+    * **integrity**: Yes (Rotating a compromised KES key is a time-consuming process, not sure if it's even possible if the KES period has not passed completely? See https://github.com/input-output-hk/cardano-node-wiki/blob/main/docs/stake-pool-operations/7_KES_period.md)
+
+#### Block diffusion
+
+Block diffusion process ensures the timely diffusion of blocks, both those produced "locally" and those received from upstream peers
+* **availability**: Yes (Not being to diffuse blocks harms a BP's SPO economic viability, and can also harm their ability to create new blocks)
+* **confidentiality**: No (block diffusion happens in the open)
+* **integrity**: Yes (partially? not sure what integrity means here)
+
+#### Block production
+
+Block production is the process of "minting" new blocks by block producers, driven by Stake-based random lottery.
+
+The mithril-signer necessarily runs on the same host than a BP because it needs access to the KES signing key
+
+* **availability**: Yes (BP is critical for SPOs revenue, and preventing a BP from producing blocks can harm SPOs capabilities to operate)
+* **integrity**: Yes (incorrect or invalid data can hamper BP capabilities)
+* **confidentiality**: Yes? (BP schedule is private information for the BP, leaking it could provide adversaries advance knowledge of schedules and lead to **grinding attacks** to attempt to manipulate nonce in the disfavour of a BP ??)
+
+#### Cardano Chain Database
+
+A cardano-node maintains an on-disk database consisting of the chain's history. This database is updated by the node when new blocks are diffused through the network, or minted, and also contains a cache of the ledger state. 
+
+Mithril signer needs needs access to _trusted_ and _up-to-date_ Chain database in order to be able to sign snapshots. 
+
+* **availability**: Yes
+* **integrity**: Yes
```

Comment:
How? I'm not entirely clear how the chain db is checked for integrity on each start, but potentially corrupted integrity could make the node have wrong ledger state / stake distributions and consequently impact block production?

Reply:
Yes, Cardano node database corruption would lead to down-time and/or impact the block production :+1:

Reply:
I added a new threat section regarding integrity of the block producer database